### PR TITLE
Jumpstart ag - Observability README

### PR DESCRIPTION
**Pull request by zaidmohd**
- Initial README
- Add reference in main README

### CODE REVIEW

File: docs/azure_jumpstart_ag/contoso_supermarket/_index.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ To get started with the "Contoso Supermarket" Jumpstart Agora scenario, we provi
...
 | Platform Engineering application config rollout with basic GitOps flow | Point of Sale (PoS), Checkout Queue Monitoring | AKS, AKS Edge Essentials, Azure Arc, Flux                                                     |
 | Developer experience with full CI/CD flow                              | Point of Sale (PoS)                            | AKS, AKS Edge Essentials, Azure Arc, Flux, GitHub Actions, Azure Container Registry           |
-| Infrastructure observability for Kubernetes and Arc-enabled Kubernetes | Infrastructure                                 | AKS, AKS Edge Essentials, Prometheus, Grafana                                                 |
+| [Infrastructure observability for Kubernetes and Arc-enabled Kubernetes](https://github.com/microsoft/azure_arc/blob/jumpstart_ag/docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md) | Infrastructure                                 | AKS, AKS Edge Essentials, Prometheus, Grafana                                                 |
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/_index.md:42]: Use a relative path for the link like in the first line

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 2]
```diff
@@ -4,4 +4,121 @@ weight: 100
 toc_hide: true
 ---
 
-# Overview
\ No newline at end of file
+# Infrastructure observability for Kubernetes and Arc-enabled Kubernetes 
+
+## Overview
+
+Infrastructure observability plays a crucial role in the success of Contoso Supermarket's cloud to edge strategy. By implementing infrastructure observability, Contoso Supermarket gains comprehensive monitoring and visualization capabilities for their Kubernetes and Arc-enabled Kubernetes environments. This empowers them to proactively monitor the health and performance of their infrastructure, identify potential issues, and make data-driven decisions to optimize their operations. With infrastructure observability, Contoso Supermarket can ensure that their cloud and edge infrastructure remain reliable, efficient, and resilient, enabling them to deliver exceptional customer experiences.
```
[the remark this answers is not in the document]
Comment (likamrat) [reply docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:11]: No need to include "Supermarket" each time, keep it simple

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 3]
```diff
@@ -4,4 +4,121 @@ weight: 100
...
+
+Prometheus is a highly efficient open-source monitoring system that collects and stores metrics from various sources in real-time. It provides a flexible query language for analyzing the collected metrics and offers robust alerting capabilities. On the other hand, Grafana is a powerful open-source data visualization and analytics platform. It allows users to create interactive and customizable dashboards to visualize the collected metrics in real-time. 
+
+By leveraging Prometheus and Grafana for infrastructure observability, Contoso Supermarket enjoys several advantages. Firstly, Prometheus's efficient data collection ensures that Contoso Supermarket can monitor crucial performance indicators and resource utilization in real-time. Secondly, Grafana provides a user-friendly interface for visualizing the collected metrics, enabling Contoso Supermarket to create interactive and customizable dashboards. These dashboards allow them to gain valuable insights into their infrastructure's health and performance, identify trends, and make informed decisions. Lastly, the combination of Prometheus and Grafana supports troubleshooting and root cause analysis.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:15]:
```suggestion
By leveraging Prometheus and Grafana for infrastructure observability, Contoso enjoys several advantages. Firstly, Prometheus's efficient data collection ensures that Contoso can monitor crucial performance indicators and resource utilization in real-time. Secondly, Grafana provides a user-friendly interface for visualizing the collected metrics, enabling Contoso to create interactive and customizable dashboards. These dashboards allow them to gain valuable insights into their infrastructure's health and performance, identify trends, and make informed decisions. Lastly, the combination of Prometheus and Grafana supports troubleshooting and root cause analysis.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 4]
```diff
@@ -4,4 +4,121 @@ weight: 100
...
+
+The Dev and Staging environments are configured with individual Prometheus and Grafana instances, while the Prod environment is configured with a central Grafana instance. This architecture allows for more granular monitoring and troubleshooting in the Dev and Staging environments, while still providing a centralized view of the infrastructure's health and performance in the Prod environment.
+
+![Observability infrastructure stack architecture diagram](./img/technology_stack.png)
+
+## Grafana dashboards
+
+Grafana's dashboards in Contoso Supermarket's implementation provide a visually appealing and customizable interface for monitoring their infrastructure. With Grafana, they can create intuitive and interactive dashboards that display key metrics and insights, empowering them to make data-driven decisions and quickly identify any issues or trends within their cloud to edge infrastructure. The following Grafana dashboards are automatically deployed as part of advanced automation to the Dev, Staging, and Prod environments:
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:27]:
```suggestion
Grafana's dashboards in Contoso's implementation provide a visually appealing and customizable interface for monitoring their infrastructure. With Grafana, they can create intuitive and interactive dashboards that display key metrics and insights, empowering them to make data-driven decisions and quickly identify any issues or trends within their cloud to edge infrastructure. The following Grafana dashboards are automatically deployed as part of advanced automation to the Dev, Staging, and Prod environments:
```

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 5]
```diff
@@ -4,4 +4,121 @@ weight: 100
...
+
+- **Kubernetes / Views / Global**: The Kubernetes Global Dashboard in Grafana offers a concise overview of your Kubernetes cluster, allowing you to quickly assess its overall health and performance. The dashboard includes panels that highlight key metrics, such as total cluster CPU, RAM, and network utilization, as well as resource usage across namespaces and nodes. Additionally, it tracks the number of resource types used in the cluster and helps detect misconfigured application resources by comparing real usage with requested and limits resources.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:29]:
```suggestion
- **Kubernetes / Views / Global**: The Kubernetes Global Dashboard in Grafana offers a concise overview of your Kubernetes cluster, allowing you to quickly assess its overall health and performance. The dashboard includes panels that highlight key metrics, such as total cluster CPU, RAM, and network utilization, as well as resource usage across namespaces and nodes. Additionally, it tracks the number of resource types used in the cluster and helps detect misconfigured application resources by comparing real usage with requested and limited resources.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 6]
```diff
@@ -4,4 +4,121 @@ weight: 100
...
+
+- **Node Exporter Full**: The Kubernetes Nodes Dashboard in Grafana provides a detailed view of node-level metrics and resources in your Kubernetes cluster. It enables you to monitor CPU and RAM usage, track pods running on each node, and identify any resource anomalies or performance issues. The dashboard also offers system-level metrics such as system load, context switches, and file descriptors, allowing for troubleshooting and optimization. Additionally, it provides insights into storage capacity, volumes, and I/O operations on the nodes, aiding in effective management of storage resources.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:31]:
```suggestion
- **Node Exporter Full**: The Kubernetes Nodes Dashboard in Grafana provides a detailed view of node-level metrics and resources in your Kubernetes cluster. It enables you to monitor CPU and RAM usage, track pods running on each node, and identify any resource anomalies or performance issues. The dashboard also offers system-level metrics such as system load, context switches, and file descriptors, allowing for troubleshooting and optimization. Additionally, it provides insights into storage capacity, volumes, and I/O operations on the nodes, aiding in the effective management of storage resources.
```

Comment (likamrat) [reply docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:31]: I would add a comment around the Freezer dashboards and point to the respective README which covers it

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 7]
```diff
@@ -4,4 +4,121 @@ weight: 100
...
+
+## Access the dashboards
+
+As an Operations team member at Contoso Supermarket, you will have access to the Grafana dashboards for infrastructure observability. These dashboards provide a comprehensive view of the health, performance, and metrics of the cloud to edge infrastructure. To access the Grafana dashboards follow the below steps.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:35]:
```suggestion
As an Operations team member at Contoso, you will have access to the Grafana dashboards for infrastructure observability. These dashboards provide a comprehensive view of the health, performance, and metrics of the cloud to edge infrastructure. To access the Grafana dashboards follow the below steps.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 8]
```diff
@@ -4,4 +4,121 @@ weight: 100
...
+
+- Connect to the Client VM `Ag-VM-Client` using the instructions in the [Deployment Guide](https://github.com/microsoft/azure_arc/blob/jumpstart_ag/docs/azure_jumpstart_ag/contoso_supermarket/deployment/_index.md#connecting-to-the-agora-client-virtual-machine).
+
+### Prod Stores
+
+- Open the Edge browser, expand Grafana in the Favorites Bar and select `Grafana Prod`.
+
+    ![Observability grafana_prod diagram](./img/grafana_prod.png)
+
+- Login using the Contoso Operator username `arcdemo` and the Windows Admin Password you provided when you created your deployment.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:45]: arcdemo?

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 9]
```diff
@@ -4,4 +4,121 @@ weight: 100
...
+
+    ![Observability grafana_prod diagram](./img/grafana_prod_dashboard_home.png)
+
+- Click `Chicago - Node Exporter Full` dashboard to review the Chicago store node metrics.
+
+    ![Observability grafana_prod diagram](./img/grafana_prod_dashboard_node.png)
+
+    ![Observability grafana_prod diagram](./img/grafana_prod_db_node01.png)
+
+- Review different panels in the dashboard to see the metrics collected from the Kubernetes cluster.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:71]:
```suggestion
- Review the different panels in the dashboard to see the metrics collected from the Kubernetes cluster.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 10]
```diff
@@ -4,4 +4,121 @@ weight: 100
...
+
+    ![Observability grafana_prod diagram](./img/grafana_prod_dashboard_seattle.png)
+
+### Staging Store
+
+- Open the Edge browser, expand Grafana in the Favorites Bar and select `Grafana Staging`.
+
+    ![Observability grafana_prod diagram](./img/grafana_staging.png)
+
+- Login using the Contoso Operator username `arcdemo` and the Windows Admin Password you provided when you created your deployment. Review the available dashboards.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:89]: arcdemo?

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md [hunk 11]
```diff
@@ -4,4 +4,121 @@ weight: 100
...
+
+As an Operations team member at Contoso Supermarket, you will have access to the Grafana dashboards for infrastructure observability. These dashboards provide a comprehensive view of the health, performance, and metrics of the cloud to edge infrastructure. To access the Grafana dashboards follow the below steps.
+
+- Connect to the Client VM `Ag-VM-Client` using the instructions in the [Deployment Guide](https://github.com/microsoft/azure_arc/blob/jumpstart_ag/docs/azure_jumpstart_ag/contoso_supermarket/deployment/_index.md#connecting-to-the-agora-client-virtual-machine).
+
+### Prod Stores
+
+- Open the Edge browser, expand Grafana in the Favorites Bar and select `Grafana Prod`.
+
+    ![Observability grafana_prod diagram](./img/grafana_prod.png)
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:43]: Take a screenshot after you cached all the icon's graphics

**Review by likamrat — CHANGES_REQUESTED** [review]
Great work @zaidmohd, super clean. 
See a few minor comments

File: docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md (hunk 10 above)
Comment (zaidmohd) [reply docs/azure_jumpstart_ag/contoso_supermarket/k8s_infra_observability/_index.md:89]: Login using the Windows Admin Username and Password you provided when you created your deployment.

**Review by likamrat — APPROVED** [review]
LGTM

**Pull request merged**